plugin/builtin/buildbaron/build_baron.go

```diff
@@ -156,16 +214,155 @@ func (bbp *BuildBaronPlugin) buildFailuresSearch(w http.ResponseWriter, r *http.
 			fmt.Sprintf("Corresponding JIRA project for %v not found", t.Project))
 		return
 	}
-	jql := taskToJQL(t, bbProj.TicketSearchProjects)
 
-	results, err := bbp.jiraHandler.JQLSearch(jql, 0, -1)
+	fallback := jiraSuggest{bbProj, bbp.jiraHandler}
+	altEndpoint := altEndpointSuggest{bbProj}
+
+	var tickets []thirdparty.JiraTicket
+	if bbProj.AlternativeEndpointEnabled {
+		tickets, err = raceSuggesters(&fallback, &altEndpoint, t)
+	} else {
+		tickets, err = fallback.Suggest(context.TODO(), t)
+	}
+
+	jql := taskToJQL(t, bbProj.TicketSearchProjects)
 	if err != nil {
 		message := fmt.Sprintf("%v: %v, %v", JIRAFailure, err, jql)
 		grip.Error(message)
 		util.WriteJSON(w, http.StatusInternalServerError, message)
 		return
 	}
-	util.WriteJSON(w, http.StatusOK, searchReturnInfo{Issues: results.Issues, Search: jql})
+	util.WriteJSON(w, http.StatusOK, searchReturnInfo{Issues: tickets, Search: jql})
+}
+
+type suggester interface {
+	Suggest(ctx context.Context, t *task.Task) ([]thirdparty.JiraTicket, error)
```

Comment:
this looks like a single function interface, i.e GetTimeout can probably be reduced to a single constant.  I'd recommend making an arrow type `type suggester func(ctx context.Context, t *task.Task) ([]thirdparty.JiraTicket, error)`, and just implementing that.

Reply:
you can get the timeout from the context `time.Now().Sub(ctx.Deadline())` and avoid the extra method for largely the same effect.

Reply:
@richardsamuels, I don't see a reason that the interface should require the JIRA endpoint the Build Baron tool's suggestion endpoint to use the same timeout for all Evergreen projects. My goal is to expose enough knobs that it is possible for the TIG team to configure the API server's usage of our service without needing to wait for a deploy in case stability is an issue or the deployment setup changes.

@tychoish, I don't understand your comment. The deadline comes from the `raceSuggesters()` function calling the `GetTimeout()` method.